pkg/resource/prunercronjob.go

```diff
@@ -0,0 +1,237 @@
+	kcorev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	batchset "k8s.io/client-go/kubernetes/typed/batch/v1beta1"
+	batchlisters "k8s.io/client-go/listers/batch/v1beta1"
+
+	imageregistryapiv1 "github.com/openshift/api/imageregistry/v1"
+	imageregistryv1listers "github.com/openshift/client-go/imageregistry/listers/imageregistry/v1"
+	"github.com/openshift/cluster-image-registry-operator/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/parameters"
+)
+
+const (
+	cronJobOwnerAnnotation = "imageregistry.openshift.io"
+)
+
+var (
+	defaultSuspend                          = false
+	defaultSchedule                         = "*/1 * * * *"
+	defaultStartingDeadlineSeconds    int64 = 60
+	defaultFailedJobsHistoryLimit     int32 = 3
+	defaultSuccessfulJobsHistoryLimit int32 = 3
+	defaultKeepTagRevisions                 = 3
+	defaultKeepYoungerThan                  = "60m"
+	defaultTolerations                      = []kcorev1.Toleration{}
+	defaultNodeSelector                     = map[string]string{}
+	defaultResources                        = kcorev1.ResourceRequirements{
+		Requests: kcorev1.ResourceList{
+			kcorev1.ResourceCPU:    resource.MustParse("100m"),
+			kcorev1.ResourceMemory: resource.MustParse("256Mi"),
+		},
+	}
+	defaultAffinity kcorev1.Affinity
+)
+
+// PrunerCronJobIsCreatedByOperator returns whether or not the PrunerCronJob was created by the Operator
+func PrunerCronJobIsCreatedByOperator(prunerCronJob *batchapi.CronJob) bool {
+	_, ok := prunerCronJob.Annotations[cronJobOwnerAnnotation]
+	return ok
+}
+
+var _ Mutator = &generatorPrunerCronJob{}
+
+type generatorPrunerCronJob struct {
+	lister       batchlisters.CronJobNamespaceLister
+	client       batchset.BatchV1beta1Interface
+	prunerLister imageregistryv1listers.ImagePrunerLister
+	cr           *imageregistryapiv1.ImagePruner
+}
+
+func newGeneratorPrunerCronJob(lister batchlisters.CronJobNamespaceLister, client batchset.BatchV1beta1Interface, prunerLister imageregistryv1listers.ImagePrunerLister, params *parameters.Globals) *generatorPrunerCronJob {
+	return &generatorPrunerCronJob{
+		lister:       lister,
+		client:       client,
+		prunerLister: prunerLister,
+	}
+}
+
+func (gcj *generatorPrunerCronJob) Type() runtime.Object {
+	return &batchapi.CronJob{}
+}
+
+func (gcj *generatorPrunerCronJob) GetGroup() string {
+	return batchapi.GroupName
+}
+
+func (gcj *generatorPrunerCronJob) GetResource() string {
+	return "batches"
+}
+
+func (gcj *generatorPrunerCronJob) GetNamespace() string {
+	return defaults.ImageRegistryOperatorNamespace
+}
+
+func (gcj *generatorPrunerCronJob) GetName() string {
+	return "image-pruner"
+}
+
+func (gcj *generatorPrunerCronJob) expected() (runtime.Object, error) {
+	cr, err := gcj.prunerLister.Get(defaults.ImageRegistryImagePrunerResourceName)
+	if err != nil {
+		return nil, err
+	}
+	gcj.cr = cr
```

Comment:
Please don't. You never know when `gcj.cr` gets changed and it makes code more complicated.

If you want to have a bunch of helpers to get values from this cr, pass this cr as an argument. Don't hide dependencies between functions.

Reply:
updated in next push